sdk/src/azure/core/az_http_response.c

```diff
@@ -158,6 +163,12 @@ AZ_NODISCARD az_result az_http_response_get_next_header(
     }
   }
 
+  if (az_span_size(ref_response->_internal.parser.remaining) == 0)
+  {
+    // avoid reading address if span is size 0
+    return AZ_ERROR_HTTP_CORRUPT_RESPONSE_HEADER;
```

Comment:
This is a more appropriate `az_result` instead of unexpected end since we don't expect something like this to happen in normal execution with a response from a service.

Good catch.